### PR TITLE
SNOW-1646980: Use docstrings folder for top-level modin.pandas IO and general functions

### PR DESCRIPTION
**Pull request by sfc-gh-joshi**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1646980

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

This PR allows separate declaration of docstrings in `modin/plugin/docstrings/io.py` and `modin/plugin/docstrings/general.py` for functions defined in the respective override files. Override functions in these namespaces should have the `_inherit_docstrings` deocrator attached; see `modin/plugin/extensions/{io_overrides,general_overrides}.py` for examples of how to do this.

`pd.read_excel` had an identical frontend implementation to upstream modin; with this docs change we can now override docstrings for top-level functions defined upstream, so this has been removed from our codebase. I've verified that the output of `help(pd.read_excel)` and generated documentation for the function have not changed.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/__init__.py [hunk 1]
```diff
@@ -97,6 +97,22 @@
 for (doc_module, target_object) in inherit_modules:
     _inherit_docstrings(doc_module, overwrite_existing=True)(target_object)
 
+# _inherit_docstrings needs a function or class as argument, so we must explicitly iterate over
+# all members of io and general
+for name in dir(modin.pandas.io):
```
Comment (sfc-gh-mvashishtha) [inline src/snowflake/snowpark/modin/plugin/__init__.py:102]: Should we rewrite these two loops as another loop over `(modin.pandas.io, modin.pandas.general)`?

**Review by sfc-gh-mvashishtha — COMMENTED** [review]
I'm leaving one style comment. Please also check if the generated docs are the same after this change.

**Comment by sfc-gh-joshi** [comment]
Did a diff on the generated HTML files to see which ones had changed, and manually verified that they matched the contents generated on `main`.

The only change is that some APIs now have links to native pandas docs at the end, like `See pandas API documentation for [pandas.isna](https://pandas.pydata.org/pandas-docs/version/2.2.2/reference/api/pandas.isna.html) for more.`

File: src/snowflake/snowpark/modin/plugin/__init__.py (hunk 1 above)
Comment (sfc-gh-joshi) [reply src/snowflake/snowpark/modin/plugin/__init__.py]: Sure, done.

**Review by sfc-gh-mvashishtha — APPROVED** [review]

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Thanks, Jonathan, for doing this.

**Pull request merged**